Example/CollectionViewToolsExample/FactoryExampleViewController.swift

```diff
@@ -0,0 +1,204 @@
+//
+//  FactoryExampleViewController.swift
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+import UIKit
+import CollectionViewTools
+
+final class FactoryExampleViewController: UIViewController {
+    
+    struct ImageData {
+        let image: UIImage
+        let description: String
+    }
+    
+    struct TextData {
+        let text: String
+        let description: String
+    }
+    
+    var data: [Any] = [ImageData(image: #imageLiteral(resourceName: "nightlife-1"), description: "First image description"),
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-2"), description: "Second image description"),
+                       TextData(text: "Fist topic", description: "First topic description"),
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-3"), description: "Third image description"),
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-4"), description: "Foursh image description"),
+                       TextData(text: "Second topic", description: "Second topic description"),
+                       TextData(text: "Third topic", description: "Third topic description"),
+                       TextData(text: "Foursh topic", description: "Foursh topic description"),
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-5"), description: "Fifth image description"),
+                       TextData(text: "Fifth topic", description: "Fifth topic description")]
+    
+    private lazy var unfoldedItemsFactory: CellItemFactory = {
+        let factory = AssociatedCellItemFactory<Any, TextCollectionViewCell>()
+        let sizeCell = TextCollectionViewCell()
+        
+        func description(data: Any) -> String {
```

Comment:
I think better to subscribe `TextData` and `ImageData` to `CustomStringConvertible` and use it instead of `Any`. Also with it that function doesn't make sense 🤔

Reply:
Improved